### PR TITLE
Removing redundant checks on property names

### PR DESCRIPTION
**Pull request by ftomassetti**
These checks were necessary before the introduction of "Internal".

Also, we remove the support for filtering properties based on visibility in DebugPrinter becomes it required a different approach to navigate properties, and because we do not consider properties' visibility elsewhere.

Tagging @loradd and @alessiostalla in case they want to take a look, otherwise I will merge in a week or so

### CODE REVIEW

File: core/src/main/kotlin/com/strumenta/kolasu/model/Printing.kt [hunk 1]
```diff
@@ -7,44 +7,36 @@ import kotlin.reflect.KVisibility.INTERNAL
...
-                it.name != "positionOverride" &&
-                it.name != "origin" &&
-                it.name != "destination" &&
-                it.name != "sourceText" &&
-                it.name != "source" &&
-                it.name != "parent"
-        }
+fun Node.relevantMemberProperties(withPosition: Boolean = false, withNodeType: Boolean = false):
+    List<KProperty1<Node, *>> {
+    val list = this.javaClass.kotlin.nodeProperties.toMutableList()
```
Comment (alessiostalla) [inline core/src/main/kotlin/com/strumenta/kolasu/model/Printing.kt:17]: I think we can simplify this to `this::class`

Comment (ftomassetti) [reply core/src/main/kotlin/com/strumenta/kolasu/model/Printing.kt]: Right, changing that

File: core/src/main/kotlin/com/strumenta/kolasu/model/Printing.kt [hunk 2]
```diff
@@ -7,44 +7,36 @@ import kotlin.reflect.KVisibility.INTERNAL
...
     var forceShowPosition: Boolean = false,
-    val hide: MutableList<String> = mutableListOf(),
-    var skipPrivateProperties: Boolean = true,
-    var skipProtectedProperties: Boolean = true,
-    var skipInternalProperties: Boolean = true,
-    var skipPublicProperties: Boolean = false
+    val hide: MutableList<String> = mutableListOf()
 )
 
 private fun KProperty1<Node, *>.hasRelevantVisibility(configuration: DebugPrintConfiguration): Boolean {
```
[the remark this answers is not in the document]
Comment (ftomassetti) [reply core/src/main/kotlin/com/strumenta/kolasu/model/Printing.kt]: True, let's remove it

[the remark this answers is not in the document]
Comment (loradd) [reply core/src/main/kotlin/com/strumenta/kolasu/model/Printing.kt]: I agree :+1:

**Pull request merged**